runtime/caml/atomic_refcount.h

```diff
@@ -16,7 +16,8 @@
 #ifndef CAML_ATOMIC_REFCOUNT_H
 #define CAML_ATOMIC_REFCOUNT_H
 
-#define CAML_INTERNALS
+#ifdef CAML_INTERNALS
```

Comment:
This doesn't need to be guarded by `CAML_INTERNALS` at all (everything the header uses is declared publicly in `config.h`). I don't know if it was used in a previous version of #11177, but `platform.h` also doesn't appear to be necessary in the header (so the `#include` should go to `runtime/bigarray.c` if `bigarray.c` really uses things from it).

Reply:
I have removed platform.h.

I propose to keep the guard, because I do not think that these would be the best tools for helping OCaml programmers with their reference-counting needs.

Reply:
That's true - and going from internals -> public can always be done if necessary in future.